irods_consortium_continuous_integration_test_hook.py

```diff
@@ -32,6 +32,7 @@ def get_package_type():
     return pt
 
 def install_test_prerequisites():
+    irods_python_ci_utilities.subprocess_get_output(['sudo', 'python3', '-m', 'pip', 'install', '--upgrade', 'pip==20.3.4'], check_rc=True)
```

Comment:
Is 20.3.4 a hard requirement?

What happens if we didn't specify a specific version?
Does it upgrade to a version that breaks testing?

Reply:
This was a change made by Alan.  I am not sure of the reasoning.

Reply:
@alanking Thoughts?

Reply:
I believe I determined this to be the earliest `pip` version that I could use to install the appropriate `minio` client. We may not need to pin the version (or at least make it something more recent) for 4.3+.

Reply:
Given that the change exists in 4-2-stable and not main, I'm fairly certain we don't need this. However, I'm not opposed to upgrading `pip` as part of this process because it sometimes causes problems if we don't.

Reply:
Pinning it doesn't sound bad. Doing that will allow us to know when the minimum version isn't sufficient. That also means we'll always be forced to know which version of pip works for the software.

Reply:
In reference to Alan's comment below, I cherry-picked and tested this on 4.3 because issue #2110 has main in the task list.

Reply:
I changed the test hook to use pip>=20.3.4.  This passed tests in the testing environment.

Note I went ahead and squashed that one commit with the original since it was so minor.